### PR TITLE
Tests for functionals

### PR DESCRIPTION
**Pull request by ksanjeevan**
From discussion in the pytorch/audio slack channel, seems like we should be doing testing for functionals. 

- ~Should we replace `test_layers` with `test_functional` and stick with that?~
- For this first pass we only want to integrate `Spectrogram`, `STFT`, `ComplexNorm` and `TimeStretch`.

(Also fixed a bug on the `phase_vocoder` on gpu)

### CODE REVIEW

**Comment by keunwoochoi** [comment]
Moving lots of things to `test_functionals.py` makes sense, but maybe not completely? There should be some test for initiating and using Layers, too. Otherwise, things look good.

**Comment by ksanjeevan** [comment]
@keunwoochoi how about `test_functional` for testing shapes and/or values, and `test_layers` for proper initialization? Maybe `test_layers` could chain some of the modules too (like spectrogram, a stretched melspectrogram, etc.)?

In any case, we'll go with just `test_functional` for Phase 1.

**Comment by keunwoochoi** [comment]
Agreed for all.

**Comment by keunwoochoi** [comment]
Hey @ksanjeevan, I think it's good to be merged. If all seems good please go ahead (and maybe make a PR to the official repo, from some point that repo would be a better place to communicate with their maintainers).

**Comment by ksanjeevan** [comment]
@keunwoochoi - Actually I'm almost done doing some changes with the proposed changes mentioned:
> how about test_functional for testing shapes and/or values, and test_layers for proper initialization? Maybe test_layers could chain some of the modules too (like spectrogram, a stretched melspectrogram, etc.)?

Just sorting a couple of bugs and I'll update this PR. Once that is done, if it looks good, 👍 to go ahead with the PR to the official repo (will update them on slack).

File: torchaudio_contrib/functional.py [hunk 1]
```diff
@@ -287,6 +288,7 @@ def amplitude_to_db(x, ref=1.0, amin=1e-7):
     Returns:
         (Tensor): same size of x, after conversion
     """
+    x = x.pow(2.)
```
Comment (ksanjeevan) [inline torchaudio_contrib/functional.py]: @keunwoochoi So I noticed that our implementation of `amplitude_to_db` was actually `power_to_db`. I added the appropiate `** 2` here and `sqrt()` for `db_to_amplitude`, as well as for their tests.

**Comment by ksanjeevan** [comment]
So I've checked some basic stuff for modules that created parameters. Maybe we want to check more? I think this makes sense. Also added tests for two combinations of modules: [`SpectrogramDb`](https://github.com/keunwoochoi/torchaudio-contrib/blob/7d1800998d559b4930d92b5d27d1a4e7209c9e9e/tests/test_layers.py#L65) (just chaining spectrogram with a db conversion) and [`MelspectrogramStretch`](https://github.com/keunwoochoi/torchaudio-contrib/blob/7d1800998d559b4930d92b5d27d1a4e7209c9e9e/tests/test_layers.py#L94) (A melspectrogram that's been stretched).

If these don't look good that's fine we can put them on hold, as long as we all agree on `test_functional` so we can proceed with Phase 1!

**Comment by keunwoochoi** [comment]
Hey, all look good. I'm not sure if we're testing *enough* but that's what torch/audio reviewers' would have opinions. Let's ship it :)

**Pull request merged**